pandas/tools/merge.py

```diff
@@ -981,16 +981,30 @@ def get_result(self):
             if self.axis == 0:
                 new_data = com._concat_compat([x._values for x in self.objs])
                 name = com._consensus_name_attr(self.objs)
-                return (Series(new_data, index=self.new_axes[0],
-                               name=name,
-                               dtype=new_data.dtype)
+
+                if com.is_sparse(new_data):
```

Comment:
this is very much like ``_concat_compat`` (well its the same idea). can we encapsulate this somewhere?